test/integration/create_classes_test.go

```diff
@@ -0,0 +1,243 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/crossplaneio/crossplane-runtime/pkg/test/integration"
+	crossplaneapis "github.com/crossplaneio/crossplane/apis"
+	clientcmd "k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/crossplaneio/stack-gcp/apis"
+	cachev1beta1 "github.com/crossplaneio/stack-gcp/apis/cache/v1beta1"
+	computev1alpha3 "github.com/crossplaneio/stack-gcp/apis/compute/v1alpha3"
+	containerv1beta1 "github.com/crossplaneio/stack-gcp/apis/container/v1beta1"
+	databasev1beta1 "github.com/crossplaneio/stack-gcp/apis/database/v1beta1"
+	storagev1alpha3 "github.com/crossplaneio/stack-gcp/apis/storage/v1alpha3"
+	"github.com/crossplaneio/stack-gcp/apis/v1alpha3"
+	"github.com/crossplaneio/stack-gcp/pkg/controller"
+)
+
+func TestCreateAllClasses(t *testing.T) {
+	cases := map[string]struct {
+		reason string
+		test   func(c client.Client) error
+	}{
+		"CreateProvider": {
+			reason: "A GCP Provider should be created without error.",
+			test: func(c client.Client) error {
+				dat, err := ioutil.ReadFile("../../examples/gcp-provider.yaml")
+				if err != nil {
+					return err
+				}
+				p := &v1alpha3.Provider{}
+				if err := yaml.Unmarshal(dat, p); err != nil {
+					return err
+				}
+
+				defer func() {
+					if err := c.Delete(context.Background(), p); err != nil {
+						t.Error(err)
+					}
+				}()
+
+				return c.Create(context.Background(), p)
+			},
+		},
+		"CreateBucketClass": {
```

Comment:
What is the thinking around having one integration test that creates all extant resource class kinds, vs having an integration test alongside each of the existing unit tests? Would it be better if we tested code closer to where it was defined, like Go unit tests do?

Reply:
This `create_classes_test.go` is a bit of a one-off. I think that most tests will not necessarily make sense to be next to a specific piece of code like unit tests. For example, I have a local test that creates a `Provider`, `GKECluster`, `CloudSQLInstance`, `KubernetesCluster` claim, `MySQLInstance` claim, and a `KubernetesApplication`. I don't think there is a specific set of code that it should be associated with.